src/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -38,13 +38,51 @@ trade-offs of pre-warming an engine.
 ### Create a FlutterEngine
 
 The proper place to create a `FlutterEngine` is specific
-to your host app. As an example, we demonstrate creating a
-`FlutterEngine`, exposed as a property, on app startup in
-the app delegate.
+to your host app. 
 
 {% samplecode engine %}
 
-{% sample Swift %}
+{% sample SwiftUI %}
+As an example, we demonstrate creating a
+`FlutterEngine` inside an `ObservableObject`, 
+and passing that into a `View` using the 
+ `environmentObject()` modifier. 
+
+ **In `MyApp.swift`:**
+ <!--code-excerpt "MyApp.swift" title-->
+ ```swift
+import SwiftUI
+import Flutter
+// Used to connect plugins (only if you have plugins with iOS platform code).
+import FlutterPluginRegistrant
+
+class FlutterDependencies: ObservableObject {
+  let flutterEngine = FlutterEngine(name: "my flutter engine")
```

Comment:
**suggestion:** Don't label the variable with the same exact label as the Object. That makes it hard to follow. 

```suggestion
  let myFlutterEngine = FlutterEngine(name: "my flutter engine")
```

Reply:
We use flutterEngine as the variable name in several examples that were already in this document and are available as demo apps in multiple places on GitHub, so I think it makes sense to leave it as is. Are you okay with that or would you still like me to change it?

Reply:
**reply:** Leave it. If you could create a new issue to address it later, that would be appreciated!

Reply:
Done!